model/models/qwen3/model.go

```diff
@@ -0,0 +1,239 @@
+package qwen3
```

Comment:
Might need to rebase for this to go away

Reply:
I cherry picked the pending qwen3 changes since I needed to use some of the updated logging to check for thinking tag tokenization. I also added some additional token logging, but if I end up beating that change getting merged then I'll separate it out into its own PR. My changes are all in one commit, so if you want to look at just my changes you'll can look at the individual commit. Not quite finished yet though (but much closer!)